src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/DotNetCoreProjectCompatibilityDetector.cs

```diff
@@ -150,8 +157,7 @@ public int OnAfterOpenProject(IVsHierarchy pHierarchy, int fAdded)
 
                         // We need to check if this project has been newly created. Our projects will implement IProjectCreationState -we can 
                         // skip any that don't
-                        IProjectCreationState projectCreationState = project.Services.ExportProvider.GetExportedValueOrDefault<IProjectCreationState>();
```

Comment:
This change was made because I could find no easy way to mock out the export provider,  To make this testable I added the `IProjectCreationInfoService` interface. The implementation of which is in `ProjectCreationInfoService.cs`

Reply:
I've take @davidwengier suggestion and just overridden the member in a test-only class